### PR TITLE
feat(route): LinkedIn Job Listing

### LINKED ISSUE #11657: LinkedIn Job Listing
### Category

Uncategorized

### Website URL

https://www.linkedin.com/jobs/search

### Website description

LinkedIn provides job listings that are good for job hunting.

### What content should be included?

We want to support some basic job queries like:
1. job titles
2. job types
3. experience levels

We want to get following information according to the query:
1. job location
2. job description
3. recruiter information

Some of those requirements are hard to achieve due to anti-scraping mechanisms of LinkedIn, they are good to implement in the future (also support for hiring managers/ recruiter search would be more appreciated).

### Additional description

_No response_

### This is not a duplicated rss request

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) and [pull requests](https://github.com/DIYgod/RSSHub/pulls) to ensure this rss proposal has not already been requested

### PR DESCRIPTION
**Pull request by BrandNewLifeJackie26**
- fix(docs): layoffs.fyi radar docs link
- feat(route): LinkedIn Job Listing
- feat(docs): add LinkedIn router docs

## 该 PR 相关 Issue / Involved issue

Close #11657

## 完整路由地址 / Example for the proposed route(s)

```routes
/linkedin/jobs/all/all
/linkedin/jobs/P-C/1-2-3/software engineer
/linkedin/jobs/P/1/software
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [x] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
添加了LinkedIn的jobs listing，`www.linkedin.com/jobs/search` 反爬严格，故使用了官方提供的API，暂时没有发现反爬的现象。后期考虑添加cookie使得用户可以爬取更多信息，像是有HR的job，以及获取详细工作信息。

LinkedIn Job Listing routes added, `www.linkedin.com/jobs/search` needs to be accessed by OAuth or cookies, so a public API is used (future improvements include adding cookies to allow descriptions and other features like fetching jobs with recruiter info)

### CODE REVIEW

**Comment by BrandNewLifeJackie26** [comment]
I have fixed the proposed routes but I don't know how to rerun the route tests, rerun it if needed, thanks!

File: docs/en/other.md [hunk 1]
```diff
@@ -150,6 +150,32 @@ RSS source in the original site is outdated.
 
 <RouteEn author="tsarlewey" example="/lever/lever" path="/lever/:domain" :paramsDesc="['Company with Lever Board']"/>
 
+## LinkedIn
+
+### Jobs
+
+<RouteEn author="BrandNewLifeJackie26" example="/jobs/C-P/1/software engineer" path="/jobs/:job_types/:exp_levels/:keywords?" :paramsDesc="['job_types: see the following table for details, use \'-\' as delimiter', 'exp_levels: see the following table for details, use \'-\' as delimiter', 'keywords']" radar="1">
```
Comment (TonyRL) [inline docs/en/other.md:157]:
```suggestion
<RouteEn author="BrandNewLifeJackie26" example="/linkedin/jobs/C-P/1/software engineer" path="/linkedin/jobs/:job_types/:exp_levels/:keywords?" :paramsDesc="['See the following table for details, use \'-\' as delimiter', 'See the following table for details, use \'-\' as delimiter', 'keywords']" radar="1">
```

File: lib/v2/linkedin/radar.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+    'linkedin.com': {
+        _name: 'linkedin',
+        '.': [
+            {
+                title: 'Job Listing',
+                docs: 'https://docs.rsshub.app/en/other.html#linkedin-jobs',
+                source: '/jobs/search/',
+                target: (params, url) => {
+                    const searchParams = new URLSearchParams(new URL('', url).search);
```
Comment (TonyRL) [inline lib/v2/linkedin/radar.js:10]:
```suggestion
                    const searchParams = new URL(url).searchParams;
```

File: lib/v2/linkedin/radar.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                    const parseRoute = (searchParam) => {
+                        if (typeof searchParam !== 'string') {
+                            return 'all';
+                        }
+                        return searchParam.split(',').join('-');
+                    };
+                    return `/jobs/${parseRoute(searchParams.get('f_JT'))}/${parseRoute(searchParams.get('f_E'))}/${searchParams.get('keywords') || ''}`;
```
Comment (TonyRL) [inline lib/v2/linkedin/radar.js:17]:
```suggestion
                    return `/linkedin/jobs/${parseRoute(searchParams.get('f_JT'))}/${parseRoute(searchParams.get('f_E'))}/${searchParams.get('keywords') || ''}`;
```

File: lib/v2/linkedin/utils.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
+const cheerio = require('cheerio');
+const { Job } = require('./models');
+
+/**
+ * Constants
+ */
+// eslint-disable-next-line no-unused-vars
```
Comment (TonyRL) [inline lib/v2/linkedin/utils.js:7]:
```suggestion
```

File: lib/v2/linkedin/utils.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+const KEYWORDS_QUERY_KEY = 'keywords';
+
+// eslint-disable-next-line no-unused-vars
```
Comment (TonyRL) [inline lib/v2/linkedin/utils.js:10]:
```suggestion
```

File: lib/v2/linkedin/utils.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+const JOB_TYPES_QUERY_KEY = 'f_JT';
+
+// eslint-disable-next-line no-unused-vars
```
Comment (TonyRL) [inline lib/v2/linkedin/utils.js:13]:
```suggestion
```

File: lib/v2/linkedin/utils.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+const JOB_TYPES = {
+    F: 'full-time',
+    P: 'part-time',
+    C: 'contract',
+};
+
+// eslint-disable-next-line no-unused-vars
```
Comment (TonyRL) [inline lib/v2/linkedin/utils.js:20]:
```suggestion
```

File: lib/v2/linkedin/utils.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+const EXP_LEVELS_QUERY_KEY = 'f_E';
+
+// eslint-disable-next-line no-unused-vars
```
Comment (TonyRL) [inline lib/v2/linkedin/utils.js:23]:
```suggestion
```

File: docs/en/other.md [hunk 9]
```diff
@@ -150,6 +150,32 @@ RSS source in the original site is outdated.
...
+
+#### `exp_levels` list
+
+| Intership | Entry Level | Associate | Mid-Senior Level | Director | All |
+| --------- | ----------- | --------- | ---------------- | -------- | --- |
+| 1         | 2           | 3         | 4                | 5        | all |
+
+For example:
+1. If we want to search software engineer jobs of all levels and all job types, use `/jobs/all/all/software engineer`
+2. If we want to search all entry level contractor/part time software engineer jobs, use `/jobs/P-C/2/software engineer`
```
Comment (TonyRL) [inline docs/en/other.md:173]:
```suggestion
1. If we want to search software engineer jobs of all levels and all job types, use `/linkedin/jobs/all/all/software engineer`
2. If we want to search all entry level contractor/part time software engineer jobs, use `/linkedin/jobs/P-C/2/software engineer`
```

**Comment by BrandNewLifeJackie26** [comment]
Thanks for all the suggested changes.

**Pull request merged**